### PR TITLE
:sparkles: feat: add PR compliance action

### LINKED ISSUE #55: 🛠 Tooling: Verify PRs match the template
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/LearningTypeScript/projects/issues?q=is%3Aissue+label%3A%22area%3A+tooling) and found none that matched my request.
- [X] This is the appropriate issue form for the tooling issue I would like to report.

### Expected

If someone sends a PR that doesn't match `PULL_REQUEST_TEMPLATE.md`'s format, a CI step should complain. I'm thinking this can be similar to the PR title check, but for the body of the PR.

### Actual

Nothing is set up now.

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by fuadop**
## PR Checklist

- [x] Addresses an existing open issue: fixes #55
- [x] That issue was marked as [accepting prs](https://github.com/LearningTypeScript/projects/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/LearningTypeScript/projects/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

@JoshuaKGoldberg, you might need to add a github secret as described on the docs of this action - https://github.com/mtfoley/pr-compliance-action#inputs.
image

### CODE REVIEW

File: .github/workflows/pr_compliance.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  pr-compliance:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: mtfoley/pr-compliance-action@main
+        with:
+          ignore-authors:
+            - dependabot
+            - dependabot[bot]
+            - all-contributors[bot]
+          title-check-enable: false
```
Comment (JoshuaKGoldberg) [inline .github/workflows/pr_compliance.yml:21]: If this bot is able to do commitlint for us, why not let it, and remove the separate commitlint action? Less configuration is nice.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Quick question on commitlint before I look deeper 😄

File: .github/workflows/pr_compliance.yml (hunk 1 above)
Comment (JoshuaKGoldberg) [reply .github/workflows/pr_compliance.yml:21]: Ping @fuadop - is this still something you have time for?

Comment (fuadop) [reply .github/workflows/pr_compliance.yml:21]: Hey @JoshuaKGoldberg, I didn't get the notification for this all this while, Let me set this to true then. It makes sense to let just one GitHub action to both.

Comment (fuadop) [inline .github/workflows/pr_compliance.yml:21]:
```suggestion
          title-check-enable: true
```

**Comment by fuadop** [comment]
Hey @JoshuaKGoldberg, can you test it out now, does it give the expected behavior?

File: .github/workflows/pr_compliance.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+name: PR Compliance
+
+on:
+  pull_request_target:
+    types: [opened, edited, reopened]
+
+# Action should have write permission to make updates to PR
```
Comment (JoshuaKGoldberg) [inline .github/workflows/pr_compliance.yml:7]: Nit: I think this is self-evident from the code. 🙂 

```suggestion
```

File: .github/workflows/pr_compliance.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  pr-compliance:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: mtfoley/pr-compliance-action@main
+        with:
+          ignore-authors:
+            - dependabot
+            - dependabot[bot]
+            - all-contributors[bot]
+          title-check-enable: true
```
Comment (JoshuaKGoldberg) [inline .github/workflows/pr_compliance.yml:21]: This means the `pull_request_title.yml` file can be deleted now, right?

Also, isn't `title-check-enable: true` the default setting?

```suggestion
```

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
This looks good so far, thanks @fuadop!

File: .github/workflows/pr_compliance.yml (hunk 3 above)
Comment (fuadop) [reply .github/workflows/pr_compliance.yml:20]: Yup, that's correct

**Review by JoshuaKGoldberg — APPROVED** [review]
Super, let's try this out! Thanks @fuadop!

**Pull request merged**